### PR TITLE
fix: default resource config

### PR DESCRIPTION
**Pull request by srodenhuis**
This PR:
- configures resource defaults for a number of apps. Design principle is NOT to specify resources in any `.gotmpl`, but instead define them in the `otomi-values` and add the defaults to the `defaults.yaml`
- optimizes resources.request for a number of apps to prevent `CPUThrottlingHigh` alerts

## Checklist

- [x] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [x] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

File: helmfile.d/snippets/defaults.yaml [hunk 1]
```diff
@@ -264,23 +264,67 @@ environments:
...
-                  cpu: 50m
+                  cpu: 500m
                   memory: 32Mi
           gitea:
             adminUsername: otomi-admin
             _rawValues: {}
+            resources:
+              gitea:
+                limits:
+                  cpu: 1
```
Comment (CasLubbers) [inline helmfile.d/snippets/defaults.yaml:278]:
```suggestion
                  cpu: "1"
```

File: helmfile.d/snippets/defaults.yaml [hunk 2]
```diff
@@ -630,8 +732,30 @@ environments:
...
+              tools:
+                limits:
+                  cpu: 600m
+                  memory: 512Mi
+                requests:
+                  cpu: 100m
+                  memory: 128Mi
           otomi-console:
             _rawValues: {}
+            resources:
```
Comment (CasLubbers) [inline helmfile.d/snippets/defaults.yaml:752]: These resources are not in the value schema, validation fails on this:
```bash
  otomi:cmd:validate-values:validateValues:error {
  otomi:cmd:validate-values:validateValues:error   keyword: 'additionalProperties',
  otomi:cmd:validate-values:validateValues:error   dataPath: '/apps/otomi-console',
  otomi:cmd:validate-values:validateValues:error   schemaPath: '#/properties/apps/properties/otomi-console/additionalProperties',
  otomi:cmd:validate-values:validateValues:error   params: { additionalProperty: 'resources' },
  otomi:cmd:validate-values:validateValues:error   message: 'must NOT have additional properties'
  otomi:cmd:validate-values:validateValues:error } 
```

**Review by CasLubbers — CHANGES_REQUESTED** [review]

**Review by CasLubbers — APPROVED** [review]

**Comment by merll** [comment]
I am getting this error 
```
2024-09-24T10:08:55.670Z otomi:cmd:validate-values:validateValues:error {
  keyword: 'additionalProperties',
  dataPath: '/apps/harbor/resources',
  schemaPath: '#/properties/apps/properties/harbor/properties/resources/additionalProperties',
  params: { additionalProperty: 'registry-controller' },
  message: 'must NOT have additional properties'
}
```

File: values-changes.yaml [hunk 3]
```diff
@@ -292,3 +292,7 @@ changes:
       - 'otomi.isHomeMonitored'
       - 'home'
       - 'dns.provider.civo'
+  - version: 27
```
Comment (merll) [inline values-changes.yaml:295]:
```suggestion
  - version: 29
```

Comment (merll) [reply values-changes.yaml:295]: Note: run `npm run migrate-values` after this change to get fixtures updated.

Comment (srodenhuis) [reply values-changes.yaml]: funny because `registry-controller` is not even supported because using `-` is not allowed ;-)

**Review by merll — APPROVED** [review]

**Pull request merged**